pinot-core/src/main/java/com/linkedin/pinot/core/io/reader/impl/v1/FixedByteChunkSingleValueReader.java

```diff
@@ -34,12 +33,20 @@
    * Constructor for the class.
    *
    * @param pinotDataBuffer Data buffer to read from
-   * @param uncompressor Chunk uncompressor
    * @throws IOException
    */
-  public FixedByteChunkSingleValueReader(PinotDataBuffer pinotDataBuffer, ChunkDecompressor uncompressor)
+  public FixedByteChunkSingleValueReader(PinotDataBuffer pinotDataBuffer)
       throws IOException {
-    super(pinotDataBuffer, uncompressor);
+    super(pinotDataBuffer);
+  }
+
+  @Override
+  public int getInt(int row) {
+    if (!isCompressed()) {
```

Comment:
why do we need this check for every access? Better to separate them into multiple implementations and get rid of the check for every access.

Reply:
Had thought about that too, it was leading to either lot of duplicate code, or adding additional level of class hierarchy. And I wasn't really strongly feeling about necessity of either.

Reply:
there will be a performance hit with this check. Create another issue to fix this later if not in this PR/

Reply:
Created issue #2223 to track this.